### ISSUE TITLE
Generate recursion separator challenge properly

### ISSUE THREAD

**Issue by lucasxia01** [AztecProtocol/barretenberg#995]
We don't generate a recursion separator challenge in the Ultra Honk prover. We need to consider the Aztec connect bug that involved this challenge.

**Comment by maramihali** [comment]
Aggregation points need to be hashed before generating the recursion separator challenge, can be done using the existing transcript object or a different one (this is something that only the verifier has to do)

**Comment by lucasxia01** [comment]
We should not be using the existing transcript for this because it should be entirely separate from the transcript of the protocol. Like we don't want this to affect the fiat-shamiring of the actual protocol. The verifier just needs to derive some random values to batch things and its not a challenge sent to the prover.

**Comment by maramihali** [comment]
Agreed, if the transcript is connected to some future steps that will be affected, but from the standpoint of security it doesn't matter what transcript we use as long as we hash them
